src/main/java/com/example/MyBatisDemo/mapstructMapper/SubjectMapper.java

```diff
@@ -0,0 +1,45 @@
+package com.example.MyBatisDemo.mapstructMapper;
+
+
+import com.example.MyBatisDemo.dto.SubjectAdditionDto;
+import com.example.MyBatisDemo.dto.SubjectDisplayDto;
+import com.example.MyBatisDemo.dto.SubjectTeacherDisplayDto;
+import com.example.MyBatisDemo.model.StudentEntity;
+import com.example.MyBatisDemo.model.SubjectEntity;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import com.example.MyBatisDemo.model.TeacherEntity;
+import com.example.MyBatisDemo.dto.*;
+import org.mapstruct.Mapping;
+import org.mapstruct.NullValuePropertyMappingStrategy;
+
+
+@Mapper(componentModel = "spring",uses = TeacherMapper.class)
+public interface SubjectMapper {
+
+    @Mapping(source = "name", target = "name")
+    @Mapping(source = "credits", target = "credits")
+    @Mapping(target = "teacher", expression = "java(setTeacherToSubject(subjectAdditionDto))")
```

Comment:
```suggestion
    @Mapping(target = "teacher.id", source="teacherId")
```
I think this should work

Reply:
added